worker/src/handlers.rs

```diff
@@ -1,13 +1,163 @@
+    request_batches_timeout: Duration,
+    // Incoming RequestBatches requests to be sent out. Uses `unreliable_send` if a target is
+    // provided, or `lucky_broadcast` if a target node count is provided.
+    rx_request_batches_rpc: Receiver<(
+        Option<NetworkPublicKey>,
+        Option<usize>,
+        WorkerBatchRequest,
+        oneshot::Sender<Result<anemo::Response<WorkerBatchResponse>>>,
+    )>,
+    /// Receive reconfiguration updates.
+    rx_reconfigure: watch::Receiver<ReconfigureNotification>,
+    // Network to use for sending requests.
+    network: P2pNetwork,
+}
+
+impl ChildRpcSender {
+    #[must_use]
+    pub fn spawn(
+        name: PublicKey,
+        id: WorkerId,
+        worker_cache: SharedWorkerCache,
+        request_batches_timeout: Duration,
+        rx_request_batches_rpc: Receiver<(
+            Option<NetworkPublicKey>,
+            Option<usize>,
+            WorkerBatchRequest,
+            oneshot::Sender<Result<anemo::Response<WorkerBatchResponse>>>,
+        )>,
+        rx_reconfigure: watch::Receiver<ReconfigureNotification>,
+        network: P2pNetwork,
+    ) -> JoinHandle<()> {
+        tokio::spawn(async move {
+            Self {
+                name,
+                id,
+                worker_cache,
+                request_batches_timeout,
+                rx_request_batches_rpc,
+                rx_reconfigure,
+                network,
+            }
+            .run()
+            .await;
+        })
+    }
+
+    async fn handle_worker_batch_responses(
+        responses: Vec<Timeout<JoinHandle<Result<anemo::Response<WorkerBatchResponse>>>>>,
+        response_channel: oneshot::Sender<Result<anemo::Response<WorkerBatchResponse>>>,
+    ) {
+        let mut results: FuturesUnordered<_> = responses.into_iter().collect();
+        let mut last_result: Result<anemo::Response<WorkerBatchResponse>> = Err(anyhow::anyhow!(
+            "no WorkerBatchResponse received (or request(s) timed out)"
+        ));
+        while let Some(result) = results.next().await {
+            match result {
+                Ok(Ok(Ok(result))) => {
+                    let _ = response_channel.send(Ok(result));
+                    return;
+                }
+                Ok(Ok(Err(e))) => last_result = Err(e),
+                _ => (),
+            }
+        }
+        let _ = response_channel.send(last_result);
+    }
+
+    async fn run(&mut self) {
+        let mut inflight_requests = FuturesUnordered::new();
+        loop {
+            tokio::select! {
+                Some((target, num_nodes, request, response_channel)) = self.rx_request_batches_rpc.recv() => {
+                    if target.is_some() && num_nodes.is_some() {
+                        panic!("cannot set both target and num_nodes");
+                    }
+                    if let Some(target) = target {
+                        match self.network.unreliable_send(target, &request) {
+                            Ok(handle) => {
```

Comment:
Could we break this handler arm out into a function for readability? (there's lots of indentation as well as rustfmt tends to break inside of macros)

Reply:
Per chat, agree it'd be better that way, but maybe skip this work if we'll be able to delete `ChildRpcSender` soonish?